src/stream/mod.rs

```diff
@@ -96,9 +97,29 @@ impl XmlWriteOptions {
     /// This may be either an `&'static str` or an owned `String`.
     ///
     /// The default is `\t`.
-    pub fn indent_string(mut self, indent_str: impl Into<Cow<'static, str>>) -> Self {
-        self.indent_str = indent_str.into();
-        self
+    ///
+    /// Since replacing `xml-rs` with `quick-xml`, the indent string has to consist of a single
+    /// repeating ascii character. This is a backwards compatibility function, prefer using
+    /// `XmlWriteOptions::with_indent`.
+    pub fn indent_string(self, indent_str: impl Into<Cow<'static, str>>) -> Self {
+        let indent_str = indent_str.into();
+        let indent_str = indent_str.as_ref();
+
+        if indent_str.is_empty() {
+            return self.with_indent(0, 0);
+        }
+
+        assert!(
```

Comment:
I'm not really sure what to do in this case - I can't imagine anyone is using a non-repeating indent string - so maybe the panic is ok.

Reply:
Considering this is a public function, this is technically a breaking change for those who use a non-repeating indent string. I'd still argue that keeping it this way and having it panic in that case is still better than making it a breaking change for everyone.

I've added a doc string for `with_indent`, which I seem to have forgotten in my first round of refactoring, and added a deprecation warning for indent_string, to reduce the chance of people getting bitten by this.